taxonomy-editor-frontend/src/pages/search/results.jsx

```diff
@@ -19,10 +20,7 @@ import DialogTitle from '@mui/material/DialogTitle';
 import Select from '@mui/material/Select';
 import ISO6391 from 'iso-639-1';
 
-const SearchResults = () => {
-    const [searchParams] = useSearchParams();
-    let query = searchParams.get('query');
-    if (query === "") query = "\"\""
+const SearchResults = ({query}) => {
     const url = API_URL+`search?query=${query}`
     const { data: nodes, isPending, isError, isSuccess, errorMessage } = useFetch(url);
     const [nodeType, setNodeType] = useState('entry'); // Used for storing node type
```

Comment:
this component is starting to accumulate a lot of states, wonder if we could remove some ...

Reply:
I have tried to remove as much of states as possible. If any other edits are possible, please do let me know.